recipes/boost/all/conanfile.py

```diff
@@ -846,9 +846,22 @@ def _create_emscripten_libs(self):
                 self.output.info(cmd)
                 self.run(cmd)
 
+    @property
+    def _is_versioned_layout(self):
+        layout = self.options.get_safe("layout")
+        return layout == "versioned" or (layout == "b2-default" and os.name == 'nt')
+
     def package_info(self):
         gen_libs = [] if self.options.header_only else tools.collect_libs(self)
 
+        if self._is_versioned_layout:
+            version_tokens = str(self.version).split(".")
```

Comment:
Unfortunately we can't use tools.Version here. By default it insert "0" when a token is not defined.